lib/Bio/EnsEMBL/DataCheck/Checks/CompareVariationFeatures.pm

```diff
@@ -0,0 +1,88 @@
+=head1 LICENSE
+
+Copyright [2018-2019] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::CompareVariationFeatures;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME           => 'CompareVariationFeatures',
+  DESCRIPTION    => 'Compare variation feature counts between two databases, categorised by seq_region name',
+  GROUPS         => ['compare_variation'],
+  DATACHECK_TYPE => 'advisory',
+  DB_TYPES       => ['variation'],
+  TABLES         => ['variation_feature']
+};
+
+sub tests {
+  my ($self) = @_;
+
+  SKIP: {
+    my $old_dba = $self->get_old_dba();
+
+    skip 'No old version of database', 1 unless defined $old_dba;
+  
+    # Check the assembly version. Skip if not the same
+    skip 'Different assemblies', 1 
+      unless $self->same_assembly($old_dba->dbc->dbname);
+    
+    my $desc = "Consistent variation feature counts between ".
+               $self->dba->dbc->dbname.' and '.$old_dba->dbc->dbname;
+
+    my ($core_dbname, $old_core_dbname);
```

Comment:
In order to get the current core db, you can use the `$self->get_dna_db` method. This is a wrapper around the core API's `dnadb` method, which does a bit of extra error-checking/housekeeping.

To get the old core db, you cannot use that method, it only works for `$self->dba`. Instead you can use `$self->get_old_dba(undef, 'core')`. The first parameter is the species (it defaults to that of `$self->dba`), the second is the database group.

With those two core DBAdaptors, you can then retrieve assembly information, and use the API to get top-level seq_regions (relying on `rank=1` for this is generally not a good idea). You then need some code to combine that with a per-seq_region_id query on variation_feature, and then to compare the results between old and new.

So the code will be more complicated, but we can't rely on the core database always being on the same server as the variation one; it's more robust and flexible if we use whatever core is available in the registry. For some pointers on cross database tests, see https://github.com/Ensembl/ensembl-datacheck/blob/master/lib/Bio/EnsEMBL/DataCheck/Checks/ForeignKeysMultiDB.pm#L48 and https://github.com/Ensembl/ensembl-datacheck/blob/master/lib/Bio/EnsEMBL/DataCheck/Checks/MetaCoord.pm#L122

Reply:
Thanks for the pointers. The datacheck has been modified to use the seq region table from the variation database and not the seq region table from the core datase. The ForeignKeysMultipleDB checks that the variation data seq_region IDs and names are consistent with the core database. If this fails, then this check should perhaps not be run. Repeating the same check on core/variation seq region ids in this check seemed like a duplication,